config/initializers/scimitar.rb

```diff
@@ -31,17 +31,52 @@
 Rails.application.config.to_prepare do
   Scimitar.service_provider_configuration = Scimitar::ServiceProviderConfiguration.new(
     patch: Scimitar::Supportable.supported,
-    authenticationSchemes: [Scimitar::AuthenticationScheme.bearer]
+    authenticationSchemes: OpenProjectScimitar::AUTHENTICATION_SCHEMES
   )
   Scimitar.engine_configuration = Scimitar::EngineConfiguration.new(
-    application_controller_mixin: ScimV2::ScimControllerMixins
+    custom_authenticator: lambda do
+      if !EnterpriseToken.allows_to?(:scim_api) || !OpenProject::FeatureDecisions.scim_api_active?
+        return handle_scim_error(Scimitar::AuthenticationError.new)
+      end
+
+      warden = request.env["warden"]
+      user = warden.authenticate(scope: :scim_v2)
+      if user == nil
```

Comment:
:green_circle: I am surprised that there is no cop for this:

```suggestion
      if user.nil?
```